lighthouse-cli/bin.ts

```diff
@@ -270,34 +198,83 @@ function handleError(err: LightHouseError) {
   }
 }
 
-function run() {
-  return initPort(cliFlags).then(() => {
-    if (cliFlags.skipAutolaunch) {
-      return lighthouseRun(urls, config, cliFlags).catch(handleError);
-    } else {
-      // because you can't cancel a promise yet
-      const isSigint = new Promise((resolve, reject) => {
-        process.on(_SIGINT, () => reject(_SIGINT));
-      });
-
-      return Promise
-        .race([launchChromeAndRun(urls, config, cliFlags), isSigint])
-        .catch(maybeSigint => {
-          if (maybeSigint === _SIGINT) {
-            return cleanup
-              .doCleanup()
-              .catch(err => {
-                console.error(err);
-                console.error(err.stack);
-              }).then(() => process.exit(_ERROR_EXIT_CODE));
-          }
-          return handleError(maybeSigint);
-        });
+/**
+ * If the requested port is 0, set it to a random, unused port.
+ */
+function initPort(flags: {port: number}): Promise<undefined> {
+  return new Promise((resolve, reject) => {
+    if (flags.port !== 0) {
+      log.verbose('Lighthouse CLI', `Using supplied port ${flags.port}`);
+      return resolve();
     }
+
+    log.verbose('Lighthouse CLI', 'Generating random port.');
+    randomPort().then(portNumber => {
```

Comment:
You're not handling the case when `randomPort()` rejects. Could result in unhandled rejections. There's no need to wrap this in a `new Promise()`, you can just return the result of that line:

```js
function initPort(flags: {port: number}): Promise<undefined> {
  if (flags.port !== 0) {
    log.verbose('Lighthouse CLI', `Using supplied port ${flags.port}`);
    return Promise.resolve();
  }

  log.verbose('Lighthouse CLI', 'Generating random port.');
  return randomPort().then(portNumber => {
    flags.port = portNumber;
    log.verbose('Lighthouse CLI', `Using generated port ${flags.port}.`);
  });
}
```

Now the caller of this function has a chance to handle possible exceptions